src/dmd/semantic3.d

```diff
@@ -1000,6 +1000,11 @@ private extern(C++) final class Semantic3Visitor : Visitor
                 if (global.params.useOut == CHECKENABLE.off)
                     fens = null;
             }
+
+            // if the function body or the in contracts contain "assert(__ctfe)" set the compile time only flag
+            funcdecl.flags |= (containsAssertCtfe(funcdecl.fbody) ? FUNCFLAG.compileTimeOnly : 0);
+            funcdecl.flags |= (containsAssertCtfe(funcdecl.frequire) ? FUNCFLAG.compileTimeOnly : 0);
```

Comment:
There's no need to check `frequire` if `fbody` already indicates a CTFE-only function.

```suggestion
            if (containsAssertCtfe(funcdecl.fbody || containsAssertCtfe(funcdecl.frequire))
                FUNCFLAG.compileTimeOnly : 0);
```

Reply:
misses a closing `)`
and I would like to avoid the branch.

Reply:
> misses a closing `)`

Fixed.

Reply:
I would still like to avoid the branch